utils/remote.cc

```diff
@@ -2457,21 +2406,50 @@ void tr_curl_easy_cleanup(CURL* curl)
     }
 }
 
-int flush(char const* rpcurl, tr_variant* benc, RemoteConfig& config)
+[[nodiscard]] long get_timeout_secs(std::string_view req)
+{
+    if (req.find(R"("method":"blocklist-update")") != std::string_view::npos)
+    {
+        return 300L;
+    }
+
+    return 60L; /* default value */
+}
+
+[[nodiscard]] std::string serialize_payload(tr_variant const* const var, RemoteConfig const& config)
+{
+    auto serde = tr_variant_serde::json();
+    serde.compact();
+    fmt::print(stderr, "{:s}:{:d} serializing payload to {:d}\n", __FILE__, __LINE__, static_cast<int>(config.network_style));
+
+    if (config.network_style == api_compat::Style::Tr5)
+    {
+        return serde.to_string(*var);
+    }
+
+    auto tmp = var->clone();
```

Comment:
Since `var` is going to be cleared at the end of `flush()`, I think it's safe to convert `var` into a non-const pointer and convert it in-place to shave off some CPU and memory overhead, instead of cloning it to `tmp`.

Looks like the logic still works even with `flush()`'s recursion.

If we go ahead with this, then https://github.com/transmission/transmission/pull/7986#discussion_r2642224722 is irrelevant.